src/passes/OptimizeInstructions.cpp

```diff
@@ -748,60 +748,56 @@ struct OptimizeInstructions
       }
       if (auto* constTrue = select->ifTrue->dynCast<Const>()) {
         if (auto* constFalse = select->ifFalse->dynCast<Const>()) {
+          Builder builder(*getModule());
           bool isPartially64bits =
             select->type == Type::i64 && select->condition->type == Type::i32;
-          auto extendIfNeeded = [=](Builder* builder, Expression* expression) {
+          auto extendIfNeeded = [&](Expression* expression) {
```

Comment:
I hope Clang/GCC has closure lifting optimization and this changes not affect to size/performance

Reply:
Yes, this should be optimized well.